Structure_Engine/Create/SectionProperties/CellularSection/CellularSectionFromBaseSection.cs

```diff
@@ -0,0 +1,228 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using BH.oM.Geometry;
+using BH.oM.Quantities.Attributes;
+using BH.oM.Spatial.ShapeProfiles;
+using BH.oM.Spatial.ShapeProfiles.CellularOpenings;
+using BH.oM.Structure.MaterialFragments;
+using BH.oM.Structure.SectionProperties;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates the maximum possible Cellular section from the provided SteelSection and opening.")]
+        [Input("baseSection", "The prismatic steel section being turned into a cellular section. The Steel section needs to have a profile of type ISectionProfile. Material from steel section will be applied to the created cellular section.")]
+        [Input("opening", "The openings in the finished cellular section. Opening setting will be used to produce as tight cut as possible on the base section, producing the cellular section with maximum possible height.")]
+        [Input("cutThickness", "Additional reduction factor on the final height of the section. Reduction based on thickness of the cutting process when producing the cellular section.", typeof(Length))]
+        [InputFromProperty("name")]
+        [Output("section", "Created cellular section.")]
+        public static CellularSection CellularSectionFromBaseSection(SteelSection baseSection, ICellularOpening opening, double cutThickness = 0, string name = "")
+        {
+            ISectionProfile baseProfile = baseSection.SectionProfile as ISectionProfile;
+
+            if (baseProfile == null)
+            {
+                Engine.Base.Compute.RecordError($"Can only create Cellular beams from SteelSections with profiles of type {nameof(ISectionProfile)}");
+                return null;
+            }
+
+            //Bring name over from baseSection to baseProfile
+            if (string.IsNullOrEmpty(baseProfile.Name) && !string.IsNullOrEmpty(baseSection.Name))
+                baseProfile.Name = baseSection.Name;
```

Comment:
Do we want to bring over `tags` and `Fragments`? I know it's not commonly used on `Profiles` but may just be worth porting over the `Base` properties for completeness?

Reply:
Not sure... But, yeah, could probably do. Will be empty for 99% of the cases, but might as well I guess.